\documentclass[a4paper, 12pt]{article}
\usepackage[T2A]{fontenc}
\usepackage[english]{babel}
\usepackage[pdftex,unicode]{hyperref}

\begin{document}
\title{\Large TandemGraph - A Tool for Modeling String Regularities \cite{biocomp}}
\author{Ramin Rakhamimov \\
        \texttt{ramin32@gmail.com} \\
        CIS 790 - Research Project\\[1cm]
        Department of Computer and Information Science, \\
        Brooklyn College of the City University of New York \\
        2900 Bedford Avenue, Brooklyn, N.Y. 11210}
\maketitle

\begin{abstract}
We have developed a tool to model string regularities. TandemGraph enables a user to study and analyze string regularities. An important application of this tool is when used to model tandem repeats \cite{tandem1,tandem2} found withing the chromosome, namely Approximate Tandem Repeats. The input for approximate tandem repeats is generated by applying the Edit Distance algorithm. A tool named TRED \cite{tred1,tred2} has been designed to yield these approximate tandem repeats.
\end{abstract}

\section{Introduction}
Once an output is generated with TRED, TandemGraph can be used to analyze the results. These repeats are drawn on a 2 dimensional plane, i.e. X is the position within the input, Y is the length of the repeat. Each tandem repeat is depicted as a triangle. The base of the triangle, along with the height, represents the length of any given repeat.  

Once TandemGraph is loaded, the following options are available to the user:
\begin{itemize}
\item Load any of the pre-filtered inputs from the in-house database.
\item Input can be zoomed in or zoomed out.
\item Input can navigated to the left or the right.
\item Lookup repeat lengths and number of errors found.
\item Switch between triangle and trepezoidal views.
\item Bring up the entire repeat alignment on demand for any individual repeat.
\end{itemize}

\section{Loading Inputs}
Using TRED we have already created a database of 20+ chromosomes that are ready to be used with TandemGraph. Upon start, TandemGraph dynamically looks up chromosome index and populates them within a file menu. The user is then required to pick a desired chromosome as an input to begin analysis. The user has an option of switching between chromosomes within the same session. Inputs can also be cleared to begin new sessions.

\section{Navigation}
After a chromosome has been successfully loaded into the application a user has several options, with which to proceed. The 2 most important navigation options are zooming and shifting. For zooming, the user may enter a specific region unto which he/she would like to zoom in on in a text box field. Positioned belowed the repeats is a rectangle designed to zoom in on regions by dragging a desired region using the naked eye. There is also a scrollbar that enables the user to continually zoom in or out with respect to the center of the current display. Along with the scrollbar there are 2 buttons (iconfied as magnifying glasses) that enable the zooming to take place at discrete intervals. When the region that is being currently displayed is equivalent in size to the width of the window, the original prefiltered chromosome data is displayed below the view.

The user may also shift the current display to the right or to the left. Shifting is done via the use of left and right navigation buttons found at the bottom of the application window. On each shift the current display is shifted in the right or left direction by half the size of the currently selected region.

At times the repeat length of any particular input may vary by a large amount. When this occurs, smaller repeats are not easliy discernable to the naked eye. We have developed and algorithm based on

\section{Statistics}
If an individual repeat is discernable within the current view it may be used to bring up various statistics about itself.By howering over the repeat with the mouse cursor, the repeat gets highlighted with its own predefined color. Once a repeat is highlighted a tooltip is displayed with the period size and errors found in calculating the approximate tandem repeat \cite{tred1,tred2}. A highlighted triangle may be clicked on, to bring up the alignment of the selected repeat.

 

\begin{thebibliography}{9}
\bibitem{biocomp}
  D. Sokol and R. Rakhamimov,
  \emph{TandemGraph: A Graphical Tool for Modeling String Regularities}.
  Proceedings of BIOCOMP '09: International Conference on Bioinformatics \& Computational Biology, 
  (Las Vegas, NV), 2009.
 
\bibitem{tandem1}
  G. Kucherov and D. Sokol. 
  \emph{Approximate Tandem Repeats}. 
  Encyclopedia of Algorithms, 2008.

\bibitem{tandem2}
  G. M. Landau, J. P. Schmidt and D. Sokol. 
  \emph{An Algorithm for Approximate Tandem Repeats}. 
  Journal of Computational Biology, 
  Volume 8, p. 1-18, 2001.

\bibitem{tred1}
  D. Sokol, G. Benson, and J. Tojeira. 
  \emph{Tandem Repeats over the Edit Distance}. 
  Bioinformatics 2007 23(2): e30-e35

\bibitem{tred2}
  D. Sokol, G. Benson, and J. Tojeira, 
  \emph{Tandem Repeats over the Edit Distance}. 
  Presented at the European Conference on Computational Biology, 
  ECCB 2006.
\end{thebibliography}

\end{document}
